frontend/src/app/modules/interventions/list/table/table.component.ts

```diff
@@ -1,44 +1,63 @@
-import { Component, Input, ViewChild, OnChanges, AfterViewInit } from '@angular/core';
+import { AfterViewInit, Component, OnInit, ViewChild } from '@angular/core';
 import { Router } from '@angular/router';
-import { MatTable, MatTableDataSource, MatPaginator } from '@angular/material';
+import { MatPaginator, MatSort, MatTable } from '@angular/material';
 
-import { Intervention } from '@shared/domain/intervention.model';
+import { Intervention, SortDirection, InterventionsFilter } from '@shared/domain/intervention.model';
+import { InterventionsDatasource } from './interventions.datasource';
+import { InterventionsService } from '../../interventions.service';
+import { tap } from 'rxjs/operators';
+import { merge } from 'rxjs';
+
+const defaultSearchParams: InterventionsFilter = {
+  pageSize: 10,
+  page: 1,
+  sortBy: 'creationDate',
+  sortDirection: SortDirection.Descending,
+};
 
 @Component({
   selector: 'app-interventions-table',
   templateUrl: './table.component.html',
   styleUrls: ['./table.component.scss'],
 })
-export class TableComponent implements OnChanges, AfterViewInit {
+export class TableComponent implements AfterViewInit, OnInit {
   displayedColumns = ['position', 'name', 'status', 'phone', 'date', 'description'];
-  dataSource: MatTableDataSource<Intervention>;
-
-  @Input() interventions: Intervention[] = [];
+  dataSource: InterventionsDatasource;
 
   @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
   @ViewChild(MatTable, { static: true }) table: MatTable<any>;
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
 
-  constructor(private router: Router) {}
-
-  ngOnChanges(changes: any) {
-    this.checkInterventions(changes.interventions.currentValue, changes.interventions.previousValue);
-  }
-
-  private checkInterventions(currentInterventions?: Intervention[], previousInterventions?: Intervention[]) {
-    if (currentInterventions !== previousInterventions) {
-      this.dataSource = new MatTableDataSource<Intervention>(currentInterventions);
-      this.dataSource.paginator = this.paginator;
-    }
-  }
+  constructor(private router: Router, private interventionsService: InterventionsService) {}
 
   showDetails(intervention: Intervention) {
     this.router.navigate(['interwencje', intervention.id]);
   }
 
+  ngOnInit() {
+    this.dataSource = new InterventionsDatasource(this.interventionsService);
+    this.dataSource.loadInterventions(defaultSearchParams);
```

Comment:
This line can be removed with fully reactive approach, see my other comment below.

Reply:
👍